### ISSUE TITLE
no-shadowed-variable false positive

### ISSUE THREAD

**Issue by michael-yx-wu** [palantir/tslint#3175]
### Bug Report

- __TSLint version__: 5.6.0
- __TypeScript version__: 2.3.3
- __Running TSLint via__: Node.js API

#### TypeScript code being linted

```ts
const { min, max } = minMax.value.transposedResults[0];
const bucket = record[independentColumnIdentifier];
const increment = (max - min) / layer.groupByInfo.numBuckets;
const xValue = min + bucket * increment;
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:latest", "tslint-react"],
  "rules": {
    "ban": {
      "severity": "error",
      "options": [
        ["Object", "getOwnPropertyNames", "use Object.keys()"],
        ["describe", "only", "should not be committed to repo"],
        ["it", "only", "should not be committed to repo"],
        ["test", "only", "should not be committed to repo"]
      ]
    },
    "no-shadowed-variable": {
      "severity": "warning"
    }
  }
}
```

#### Actual behavior

Trans that `min`, `max` are being shadowed when they're just being used to assign other variables. If I comment out the 3rd and 4th lines, the warning goes away.

#### Expected behavior

No warning

**Comment by ajafff** [comment]
I cannot reproduce the bug with the code snippet from above. Is that the whole code that is needed to reproduce it?

**Comment by michael-yx-wu** [comment]
Turns out I was shadowing lodash imports further up in the file. Thanks @adidahiya for the help here. Sorry for the false alarm!
